### PR TITLE
ci: Add flaky marker to LTS tests

### PR DESCRIPTION
**Pull request by ko3n1g**
# What does this PR do ?

:warning: For major changes (either in lines of code or in its impact), please make sure to first share discuss a design-doc with the team.  

## Contribution process

```mermaid
flowchart LR
    A[Pre-checks] --> B[PR Tests]
    subgraph Code Review/Approval
        C1[Expert Review] --> C2[Final Review]
    end
    B --> C1
    C2 --> D[Merge]
```

### Pre-checks

- [ ] I want this PR in a versioned release and have added the appropriate Milestone (e.g., `Core 0.8`)
- [ ] I have added relevant unit tests
- [ ] I have added relevant functional tests
- [ ] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
- [ ] I have added relevant documentation
- [ ] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

### Code review

The following process is enforced via the CODEOWNERS file for changes into `megatron/core`. For changes outside of `megatron/core`, it is up to the PR author whether or not to tag the Final Reviewer team.

<details>
<summary>For MRs into `main` branch</summary>

#### (Step 2): Collect the expert reviewers reviews

1. Attach the `Expert Review` label when your PR is ready for review.
2. GitHub auto-assigns expert reviewers based on your changes. They will get notified and pick up your PR soon.

:warning: Only proceed to the next step once all reviewers have approved, merge-conflict are resolved and the CI is passing.  
Final Review might get declined if these requirements are not fulfilled.

#### (Step 3): Final Review

1. Add `Final Review` label
2. GitHub auto-assigns final reviewers based on your changes. They will get notified and pick up your PR soon.

#### (Optional Step 4): Cherry-pick into release branch

If this PR also needs to be merged into `core_r*` release branches, after this PR has been merged, select `Cherry-pick` to open a new PR into the release branch.

</details>

<details>
<summary>For MRs into `dev` branch</summary>
The proposed review process for `dev` branch is under active discussion.

MRs are mergable after one approval by either `eharper@nvidia.com` or `zijiey@nvidia.com`.
</details>

### Merging your PR

Any member of [core-adlr](https://github.com/orgs/teams/NVIDIA/core-adlr) and [`core-nemo`](https://github.com/orgs/teams/NVIDIA/core-nemo) will be able to merge your PR.

### CODE REVIEW

File: tests/unit_tests/data/test_bin_reader.py [hunk 1]
```diff
@@ -142,6 +142,7 @@ def read(self, path, byte_range):
 setattr(msc, "resolve_storage_client", _msc_resolve_storage_client)
 
 
+@pytest.mark.flaky
 @pytest.mark.flaky
```
Comment (pablo-garay) [inline tests/unit_tests/data/test_bin_reader.py:145]: two same lines for this one

File: tests/unit_tests/data/test_preprocess_data.py [hunk 2]
```diff
@@ -213,6 +213,7 @@ def bert_vocab(odir):
     return path
 
 
+@pytest.mark.flaky
 @pytest.mark.flaky
```
Comment (pablo-garay) [inline tests/unit_tests/data/test_preprocess_data.py:216]: two same (duplicated) lines for this one

File: tests/unit_tests/dist_checkpointing/test_fully_parallel.py [hunk 3]
```diff
@@ -331,7 +331,8 @@ def test_load_distribution(self, parallelize_within_dp, tmp_path_dist_ckpt):
 
     @pytest.mark.parametrize('state_dict_device', ['cpu', 'cuda'])
     @pytest.mark.flaky
```
Comment (pablo-garay) [inline tests/unit_tests/dist_checkpointing/test_fully_parallel.py:333]: two same lines for this one

File: tests/unit_tests/distributed/test_grad_sync_with_expert_parallel.py [hunk 4]
```diff
@@ -109,6 +109,7 @@ def get_moe_model_and_buffers(
 @pytest.mark.parametrize("etp_size", [1, 2])
 @pytest.mark.parametrize("num_distributed_optimizer_instances", [1, 2])
 @pytest.mark.flaky
```
Comment (pablo-garay) [inline tests/unit_tests/distributed/test_grad_sync_with_expert_parallel.py:111]: two same lines for this one

File: tests/unit_tests/test_parallel_state.py [hunk 5]
```diff
@@ -235,6 +236,7 @@ def test_different_initialize_order_consistency(src_tp_pp, ep_size):
     [((1, 2), 1), ((1, 4), 1), ((2, 2), 1), ((1, 2), 2), ((1, 4), 2), ((2, 2), 2)],
 )
 @pytest.mark.flaky
```
Comment (pablo-garay) [inline tests/unit_tests/test_parallel_state.py:240]: two same lines for this one
